Metadata.Rmd

```diff
@@ -0,0 +1,660 @@
+    based on "The Grammar of Graphics". You provide the data, tell 'ggplot2'
+    how to map variables to aesthetics, what graphical primitives to use,
+    and it takes care of the details.
+```
+
+A good title and description are important, especially if you plan to release your package to CRAN, because they appear on the CRAN download page as follows:
+
+<!-- TODO: I know my hacky diagram might need replacement, but at least it's more up-to-date. -->
+
+```{r cran-package-page}
+#| echo = FALSE, out.width = "100%",
+#| fig.cap = "The CRAN page for ggplot2, highlighting Title and Description."
+knitr::include_graphics("diagrams/cran-package-ggplot2.png")
+```
+
+You'll notice that `Description` only gives you a small amount of space to describe what your package does.
+This is why it's so important to also include a `README.md` file that goes into much more depth and shows a few examples.
+You'll learn about that in [README.md](#readme).
+
+## Author: who are you? {#description-authors}
+
+To identify the package's author, and whom to contact if something goes wrong, use the `Authors@R` field.
+This field is unusual because it contains executable R code rather than plain text. Here's an example:
+
+```yaml
+Authors@R: person("Hadley", "Wickham", email = "hadley@rstudio.com",
+  role = c("aut", "cre"))
+```
+
+```{r}
+person("Hadley", "Wickham", email = "hadley@rstudio.com", 
+  role = c("aut", "cre"))
+```
+
+This command says that both the author (aut) and the maintainer (cre) is Hadley Wickham, and that his email address is `hadley@rstudio.com`.
+The `person()` function has four main inputs:
+
+* The name, specified by the first two arguments, `given` and `family` (these
+  are normally supplied by position, not name).
+  In English cultures, `given` (first name) comes before `family` (last name).
+  In many cultures, this convention does not hold.
+  For a non-person entity, such as "R Core Team" or "RStudio", use the `given`
+  argument (and omit `family`).
+  
+* The `email` address.
+  It's important to note that this is the address CRAN uses to let you know
+  if your package needs to be fixed in order to stay on CRAN.
+  Make sure to use an email address that's likely to be around for a while.
+  CRAN policy requires that this be for a person, as opposed to, e.g., a
+  mailing list.
+
+* One or more three letter codes specifying the `role`.
+  There are four important roles:
+
+    * `cre`: the creator or maintainer, the person you should bother 
+      if you have problems.
+      
+    * `aut`: authors, those who have made significant contributions to the 
+      package.
+    
+    * `ctb`: contributors, those who have made smaller contributions, like 
+      patches.
+    
+    * `cph`: copyright holder. This is used if the copyright is held by someone 
+      other than the author, typically a company (i.e. the author's employer).
+    
+    (The [full list of roles](https://www.loc.gov/marc/relators/relaterm.html) is
```

Comment:
Just in case you need it, I discovered this recently:

``` r
db <- getNamespace("utils")$MARC_relator_db
db <- db[db$usage != "",]
tibble::as_tibble(db)
#> # A tibble: 11 × 4
#>    term             code  description                   usage                   
#>    <chr>            <chr> <chr>                         <chr>                   
#>  1 Author           aut   A person, family, or organiz… Use for full authors wh…
#>  2 Compiler         com   A person, family, or organiz… Use for package maintai…
#>  3 Contractor       ctr   A person or organization rel… Use for authors who hav…
#>  4 Contributor      ctb   A person, family or organiza… Use for authors who hav…
#>  5 Copyright holder cph   A person or organization to … Use for all copyright h…
#>  6 Creator          cre   A person or organization res… Use for the package mai…
#>  7 Data contributor dtc   A person or organization tha… Use for persons who con…
#>  8 Funder           fnd   A person or organization tha… Use for persons or orga…
#>  9 Reviewer         rev   A person or organization res… Use for persons or orga…
#> 10 Thesis advisor   ths   A person under whose supervi… If the package is part …
#> 11 Translator       trl   A person or organization who… If the R code is merely…
```

<sup>Created on 2021-12-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>